pkg/agent/openflow/multicast.go

```diff
@@ -121,6 +135,84 @@ func (f *featureMulticast) multicastOutputFlow(cookieID uint64) binding.Flow {
 		Done()
 }
 
+// igmpPktInFlows generates the flow to load CustomReasonIGMPRegMark to mark the IGMP packet in MulticastIGMPEgressTable
+// and sends it to antrea-agent.
+func (f *featureMulticast) igmpPktInFlows(reason uint8) []binding.Flow {
+	table := MulticastIGMPEgressTable
+	if !f.enableAntreaPolicy {
+		table = MulticastRoutingTable
+	}
+	flows := []binding.Flow{
+		// Set a custom reason for the IGMP packets, and then send it to antrea-agent and forward it normally in the
+		// OVS bridge, so that the OVS multicast db cache can be updated, and antrea-agent can identify the local multicast
+		// group and its members in the meanwhile.
+		// Do not set dst IP address because IGMPv1 report message uses target multicast group as IP destination in
+		// the packet.
+		table.ofTable.BuildFlow(priorityHigh).
+			Cookie(f.cookieAllocator.Request(f.category).Raw()).
+			MatchProtocol(binding.ProtocolIGMP).
+			MatchRegMark(FromLocalRegMark).
+			Action().LoadRegMark(CustomReasonIGMPRegMark).
+			Action().SendToController(reason).
+			Action().Normal().
+			Done(),
+	}
+	return flows
+}
+
+// localMulticastForwardFlows generates the flow to forward multicast packets with OVS action "normal", and outputs
+// it to Antrea gateway in the meanwhile, so that the packet can be forwarded to local Pods which have joined the Multicast
+// group and to the external receivers. For external multicast packets accessing to the given multicast IP also hits the
```

Comment:
I do not understand "For external multicast packets accessing to the given multicast IP". Do you mean "Multicast packets from external network"?

hits -> hit